ledger/catchpointtracker_test.go

```diff
@@ -1185,104 +1189,128 @@ func TestCalculateCatchpointRounds(t *testing.T) {
 func TestCatchpointFirstStageInfoPruning(t *testing.T) {
 	partitiontest.PartitionTest(t)
 
-	// create new protocol version, which has lower lookback
-	testProtocolVersion :=
-		protocol.ConsensusVersion("test-protocol-TestFirstStageInfoPruning")
-	protoParams := config.Consensus[protocol.ConsensusCurrentVersion]
-	protoParams.CatchpointLookback = 32
-	protoParams.EnableCatchpointsWithSPContexts = true
-	config.Consensus[testProtocolVersion] = protoParams
-	defer func() {
-		delete(config.Consensus, testProtocolVersion)
-	}()
-
-	accts := []map[basics.Address]basics.AccountData{ledgertesting.RandomAccounts(20, true)}
-
-	ml := makeMockLedgerForTracker(t, false, 1, testProtocolVersion, accts)
-	defer ml.Close()
+	tempDir := t.TempDir()
+	var tests = []struct {
+		hotPath  string
+		coldPath string
+	}{
+		{"", ""},
+		{"hot", "cold"},
+	}
 
-	cfg := config.GetDefaultLocal()
-	cfg.CatchpointInterval = 4
-	cfg.CatchpointTracking = 2
-	ct := newCatchpointTracker(t, ml, cfg, ".")
-	defer ct.close()
+	for _, test := range tests {
+		var testName string
+		if test.hotPath == test.coldPath {
+			testName = "dirs=same"
+		} else {
+			testName = "dirs=different"
+		}
+		t.Run(testName, func(t *testing.T) {
```

Comment:
the same test as before but nested with equal (original) and different hot/cold dirs. Dirs assignment is done inside `newCatchpointTrackerWithPaths` as part of initialization instead of initialized `ct`'s properties modification.

Reply:
Am I misreading this, or is tempDir assumed to contain both the hot and cold paths (this may just be testing logic)? I would expect there to be a tempdir on the hot directory but this might just be a sanity check.

Reply:
yes, the original test used the same dir for both hot and cold, and it was `t.TempDir()`.
Updated version additionally exercises different hot and cold.